### PR TITLE
add keycloak (#1215)

### PR DESCRIPTION
**Pull request by bill-transue**
* Add Keycloak Operations and Specs
* Update .docker to re-enable dev dockerfile build and docker compose
* Update .docker to re-enable dev dockerfile build and docker compose
* Add find and sign-in operations. Rename token methods
* Remove prettier.rc
* Add User create operation.
Update User model to support keycloak
Updated User model email regex to align with AcaEntities
Remove non UTF-8 char from authorization concern

* WIP on restoring Docker config

* Add polypress client keycloak config

* updating gems

* Updated dev Dockerfile to reference local folders
Updated Dockerfile to install node and yarn

* WIP on Docker

* Change enroll boot command.
Disable remote_event_listener

* Fix Development RabbitMQ Dockerfile

* Clean Dockerfile build and docker compose

* Exclude local gem builds

* WIP on shared context

* Exclude bash_history

* Fix mongoid matcher error

* WIP on Accounts
Add shared context
Update find method
Add map_attributes operation

* WIP on Accounts find operation

* Updated account create and fixes for map_attributes operation

* Updated map_attriubtes to handle arrays
Fixed find_spec before/after tags

* Add spec to find operation
Add stubs and action items to account shared contect

* Add ResetPassword operation
WIP on spec

* Added keycloak support to user accounts datatable

* WIP debugging reset_password

* Add keycloak lock/unlock, reset password views

* Update User actions for Keycloak integration

* Fix rubocop errors

* Fix mongo server for tests

* Fix for the contribution model factory

* Fixed keycloak specs with VCR

* Fixed spec failures around contribution units

* reverted mongoid, mongo, mongoid-rspec gem updates

* Added missing cassettes for users

* Downgrade cucumber

* Fix sponsored benefit gems

* updated cassettes and user create operation

* Condition datatable search method

* Fix FAA operation

* Fix allow_instance_of

* Fixed reset password spec

* Added account update operation and specs

* Add change password views & new accounts datatable

* Revert changes to UserAccountDatatable

* Revert changes to UserDatatableQuery

* Fix rubocop

* Fixed user accounts datatable actions and updated display

* rubocop fixes

* fixed rubocop error

* Fix AutoTransmitMonthlyBenefitSponsors spec

* Fix rubocop

* Fix User spec

* Fix FamilyApplicationCompleted

* Fix FamilyApplicationCompleted spec

* Pagination fixes for accounts data table

* Added count_all scope to account>find operation

* Fix Migration spec

* Fix _consumer_brokers_widget spec

* Fix for SelfServiceFactory

* Fix the fix for SelfServiceFactory

Co-authored-by: Dan Thomas <dan@ideacrew.com>
Co-authored-by: Raghuram Ganjala <raghuramg83@gmail.com>
Co-authored-by: Bill Transue <transue@gmail.com>

### CODE REVIEW

File: app/controllers/saml_controller.rb [hunk 1]
```diff
@@ -16,7 +16,11 @@ def login
 
     if response.is_valid? && response.name_id.present?
       username = response.name_id.downcase
```
Comment (ipublic) [inline app/controllers/saml_controller.rb:18]: My understanding is response.name_id will hold the keycloak id value when keycloak is ident mgr.  this logic doesn't seem to reflect that?

File: app/domain/operations/accounts/forgot_password.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
```
Comment (ipublic) [inline app/domain/operations/accounts/forgot_password.rb:1]: This operation needs dev testing against new KC realm config

File: app/domain/operations/accounts/sign_in.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
```
Comment (ipublic) [inline app/domain/operations/accounts/sign_in.rb:1]: Delete this unused code?

**Review by ipublic — CHANGES_REQUESTED** [review]

File: app/controllers/saml_controller.rb (hunk 1 above)
Comment (bill-transue) [reply app/controllers/saml_controller.rb:18]: You're right, it belongs in the `else` condition below. Thanks.

File: app/domain/operations/accounts/sign_in.rb (hunk 3 above)
Comment (bill-transue) [reply app/domain/operations/accounts/sign_in.rb:1]: Will do.

**Review by ipublic — APPROVED** [review]

**Pull request merged**